### ISSUE TITLE
`reportImplicitOverride` reports errors as `reportGeneralTypeIssues`

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#4824]
Given the following code with `reportImplicitOverride` and `reportUnnecessaryTypeIgnoreComment` both set to `"error"`:

```python
class Base:
    def do_something(self) -> None:
        print('here')


class Thing(Base):
    def do_something(self) -> None:
        print('here instead')
```

The error at line 7 is reported as:

```
test.py:7:9 - error: Method "do_something" is not marked as override but is overriding a method in class "Base" (reportGeneralTypeIssues)
```

If I add `# pyright: ignore[reportImplicitOverride]` to the end of the line to suppress it, I get second error:

```
test.py:7:9 - error: Method "do_something" is not marked as override but is overriding a method in class "Base" (reportGeneralTypeIssues)
test.py:7:56 - error: Unnecessary "# pyright: ignore" rule: "reportImplicitOverride"
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
1.1.300

**Comment by erictraut** [comment]
This is included in pyright 1.1.301, which I just published. It will also be included in a future release of pylance.
